src/agent/multitype_queue/include/imultitype_queue.hpp

```diff
@@ -61,15 +62,16 @@ class IMultiTypeQueue
      * @brief Retrieves the next message from the queue asynchronously.
      *
      * @param type The type of the queue to use as the source.
-     * @param messageQuantity The quantity of messages to return.
+     * @param messageQuantity In bytes or in number of messages.
      * @param moduleName The name of the module requesting the message.
      * @param moduleType The type of the module requesting the messages.
      * @return boost::asio::awaitable<std::vector<Message>> Awaitable object representing the next N messages.
      */
-    virtual boost::asio::awaitable<std::vector<Message>> getNextNAwaitable(MessageType type,
-                                                                           int messageQuantity,
-                                                                           const std::string moduleName = "",
-                                                                           const std::string moduleType = "") = 0;
+    virtual boost::asio::awaitable<std::vector<Message>>
+    getNextNAwaitable(MessageType type,
+                      std::variant<const int, const MessageSize> messageQuantity,
```

Comment:
Why don't you define two functions like `getNextNAwaitable` and `getNextBytesAwaitable`? This will make things easier at runtime since we know what we want to retrieve everywhere this function is called.